setup.py

```diff
@@ -100,14 +100,14 @@
     keywords="",
     install_requires=[
         "PyJWT>=2.0.0 ,<2.4.0",
-        "httpx>=0.15.0 ,<0.23.0",
+        "httpx>=0.15.0 ,<0.24.0",
         "pycryptodome==3.10.*",
         "jsonschema==3.2.0",
         "tldextract==3.1.0",
-        "asgiref==3.4.1",
+        "asgiref>=3.4.1,<4",
```

Comment:
- `asgiref`: The only place where we are using it is `django_middleware.py` as: `from asgiref.sync import async_to_sync`
- `httpx`: We are using it only for `from httpx import AsyncClient, HTTPStatusError`
- `cryptography`: We are using it only for `from cryptography.hazmat.primitives.asymmetric.rsa import RSAPublicKey` 

For asgiref, all the things mentioned in the changelog related to `async_to_sync_wrapper` look like feature improvements and not something that will affect the existing behaviour. (try doing Ctrl+F)

For httpx, All the things mentioned in [changelog](https://github.com/encode/httpx/blob/master/CHANGELOG.md) related to `AsyncClient` and `HTTPStatusError` don't seem to affect our way of using it. We are just using it for `async_to_sync(st.middleware)` (try doing Ctrl+F)

For cryptography, The [changelog](https://github.com/pyca/cryptography/blob/37.0.4/CHANGELOG.rst) seems a little complicated but since RSA encryption logic is clearly defined in cryptography theory, so imo, we just need to make sure that imports and types works fine. So I tried installing the latest version `cryptography==37.4` and ran `make check-lint` and even the tests were successful

But, I see that the changelog said this for 37.0.0:

```
BACKWARDS INCOMPATIBLE: Removed signer and verifier methods from the public key and private key classes. 
These methods were originally deprecated in version 2.0, but had an extended deprecation timeline due to usage. 
Any remaining users should transition to sign and verify.
```
Before this (in 36.0.0) these funcs were just deprecated (not removed). That's why I felt it's safer to go for `<37.0.0`